src/components/charts/ohlcChart.tsx

```diff
@@ -2,31 +2,32 @@
 /* eslint-disable -- disabling this file as this was created before our strict rules */
 // src/components/charts/ohlcChart.tsx
 
-import React, { useEffect, useState } from "react";
-import { VStack, Text, Button, ButtonGroup } from "@chakra-ui/react";
-import { Token } from "@/utils/types/token";
-import { LoadingSpinner } from "../util/loadingSpinner";
-import ReactECharts from "echarts-for-react";
-import { format } from "date-fns";
+import React, { useEffect, useState, useRef } from 'react';
+import { VStack, Text, Button, ButtonGroup, Flex } from '@chakra-ui/react';
+import useComponentSize from '@rehooks/component-size';
+import { Token } from '@/utils/types/token';
+import { LoadingSpinner } from '../util/loadingSpinner';
+import ReactECharts from 'echarts-for-react';
+import { format } from 'date-fns';
 
 interface OHLCChartProps {
   asset1Token: Token;
   asset2Token: Token;
 }
 const OHLCChart = ({ asset1Token, asset2Token }: OHLCChartProps) => {
+  const wrapperRef = useRef();
+  const { width } = useComponentSize(wrapperRef);
   const [isLoading, setIsLoading] = useState(true);
   const [isOHLCDataLoading, setIsOHLCDataLoading] = useState(true);
   const [isTimestampsLoading, setIsTimestampsLoading] = useState(true);
   const [ohlcData, setOHLCData] = useState([]); // [{open, high, low, close, directVolume, swapVolume, height}]
   const [originalOHLCData, setOriginalOHLCData] = useState([]); // [{open, high, low, close, directVolume, swapVolume, height}
   const [blockToTimestamp, setBlockToTimestamp] = useState<Record<string, string>>({}); // {height: timestamp}
   const [error, setError] = useState<string | undefined>(undefined); // [error message]
-  const [chartData, setChartData] = useState<
-    [string, number, number, number, number][]
-  >([]); // [[date, open, close, low, high]]
+  const [chartData, setChartData] = useState<[string, number, number, number, number][]>([]); // [[date, open, close, low, high]]
```

Comment:
_(not for later)_
At some point, would be great to convert this data structure to an object versus ordered array.